test/e2e_node/e2e_node_suite_test.go

```diff
@@ -117,10 +116,28 @@ var _ = BeforeSuite(func() {
 
 	// Reference common test to make the import valid.
 	commontest.CurrentSuite = commontest.NodeE2E
+
+	data, err := shared.Encode()
+	Expect(err).NotTo(HaveOccurred())
+
+	return data
+}, func(data []byte) {
+	// Set the shared context got from the synchronized initialize function
+	shared := &SharedContext{}
+	Expect(shared.Decode(data)).To(Succeed())
+	context = *shared
```

Comment:
What's going on here? Why do you have a separate function, and serialize the shared context in between?